ptp/ptp4u/server/server.go

```diff
@@ -151,6 +154,12 @@ func (s *Server) Start() error {
 
 	// Wait for ANY gorouine to finish
 	wg.Wait()
+
+	// graceful shutdown
+	if s.stop {
+		return nil
+	}
```

Comment:
Wait, can't we check the context.Done instead?

Reply:
Well, the `handleSigterm()` is calling the `Drain()`. So technically during shutdown we must have `Conext.Done()`. It may be better than sticking a boolean for this particular case...

Reply:
Still, the service may be drained for another reason and trigger a non graceful shutdown (no sigterm). So I think reusing it will hide some potential issues.

Reply:
Wait I am confused. `handleSigterm()` is called on `SIGTERM`. And first we drain it, then we stop. If other signal comes, this boolean will not be set anyway and will lead to a panic like before.

Reply:
The server may go to shutdown if any of the following routines ends:
* general listener
* event listener
* drain check
* handle sighup

In those cases regardless of the drain status we keep the same behavior and return an error.

We only avoid returning an error when handling sigterm.